### DISCUSSION TITLE
Vendor ZYXH - TS0601_switch_12 - Not Supported

### DISCUSSION THREAD

**Discussion by Franckybel** [Koenkk/zigbee2mqtt#18339] [Devices]
Hello,

I have received a new device ZYXH with 12 switches, it looks very efficient but not recognized by zigbee2mqtt.
![image](https://github.com/Koenkk/zigbee2mqtt/assets/53462598/c48c9947-01f9-4000-ab3e-aed35825030f)
https://fr.aliexpress.com/item/1005005605518721.html

I see that a same device from ZYXH with 8 switches is recognized by  a zigbee2mqtt
https://www.zigbee2mqtt.io/devices/TS0601_switch_8.html

I'm not really familiar wth all settings (I'm a standard user ;)), How I can integrate this new device.
I have started the procedure to integrate unsupported device but unfortunately, I have no exposed added info after restart with debug mode.
![image](https://github.com/Koenkk/zigbee2mqtt/assets/53462598/98c59e7f-5705-49bf-9255-ee788dd7f937)
![image](https://github.com/Koenkk/zigbee2mqtt/assets/53462598/303af55b-c858-4eb7-ba71-087e61b590dd)

`const fz = require('zigbee-herdsman-converters/converters/fromZigbee');
const tz = require('zigbee-herdsman-converters/converters/toZigbee');
const exposes = require('zigbee-herdsman-converters/lib/exposes');
const reporting = require('zigbee-herdsman-converters/lib/reporting');
const extend = require('zigbee-herdsman-converters/lib/extend');
const e = exposes.presets;
const ea = exposes.access;

const definition = {
    zigbeeModel: ['TS0601'], // The model ID from: Device with modelID 'lumi.sens' is not supported.
    model: 'TS0601', // Vendor model number, look on the device for a model number
    vendor: 'ZYXH', // Vendor of the device (only used for documentation and startup logging)
    description: 'Tuya-Commutateur 12 voies', // Description of the device, copy from vendor site. (only used for documentation and startup logging)
    fromZigbee: [], // We will add this later
    toZigbee: [], // Should be empty, unless device can be controlled (e.g. lights, switches).
    exposes: [e.battery(), e.temperature(), e.humidity()], // Defines what this device exposes, used for e.g. Home Assistant discovery and in the frontend
};

module.exports = definition;`

**Comment by Franckybel** [comment]
Hi Team,

I have successfully integrated this new hardware following unsupported Tuya device, here the content

const fz = require('zigbee-herdsman-converters/converters/fromZigbee');
const tz = require('zigbee-herdsman-converters/converters/toZigbee');
const exposes = require('zigbee-herdsman-converters/lib/exposes');
const reporting = require('zigbee-herdsman-converters/lib/reporting');
const extend = require('zigbee-herdsman-converters/lib/extend');
const e = exposes.presets;
const ea = exposes.access;
const tuya = require('zigbee-herdsman-converters/lib/tuya');

const definition = {
    // Since a lot of TuYa devices use the same modelID, but use different datapoints
    // it's necessary to provide a fingerprint instead of a zigbeeModel
    fingerprint: [
        {
            // The model ID from: Device with modelID 'TS0601' is not supported
            // You may need to add \u0000 at the end of the name in some cases
            modelID: 'TS0601',
            // The manufacturer name from: Device with modelID 'TS0601' is not supported.
            manufacturerName: '_TZE204_dqolcpcp',
        },
    ],
    model: 'TS0601_12Swtichs',
    vendor: 'TuYa',
    description: '12 Switch Tuya ZYXH',
    fromZigbee: [tuya.fz.datapoints],
    toZigbee: [tuya.tz.datapoints],
    onEvent: tuya.onEventSetTime, // Add this if you are getting no converter for 'commandMcuSyncTime'
    configure: tuya.configureMagicPacket,
     exposes: [
            tuya.exposes.switch().withEndpoint('l1'),
            tuya.exposes.switch().withEndpoint('l2'),
            tuya.exposes.switch().withEndpoint('l3'),
            tuya.exposes.switch().withEndpoint('l4'),
            tuya.exposes.switch().withEndpoint('l5'),
            tuya.exposes.switch().withEndpoint('l6'),
            tuya.exposes.switch().withEndpoint('l7'),
            tuya.exposes.switch().withEndpoint('l8'),
			tuya.exposes.switch().withEndpoint('l9'),
			tuya.exposes.switch().withEndpoint('l10'),
			tuya.exposes.switch().withEndpoint('l11'),
			tuya.exposes.switch().withEndpoint('l12'),

        ],
  endpoint: (device) => {
            return {'l1': 1, 'l2': 1, 'l3': 1, 'l4': 1, 'l5': 1, 'l6': 1, 'l7': 1, 'l8': 1,'l9': 1,'l10': 1,'l11': 1,'l12': 1};
        },
   meta: {
            multiEndpoint: true,
            tuyaDatapoints: [
                [1, 'state_l1', tuya.valueConverter.onOff],
                [2, 'state_l2', tuya.valueConverter.onOff],
                [3, 'state_l3', tuya.valueConverter.onOff],
                [4, 'state_l4', tuya.valueConverter.onOff],
                [5, 'state_l5', tuya.valueConverter.onOff],
                [6, 'state_l6', tuya.valueConverter.onOff],
                [0x65, 'state_l7', tuya.valueConverter.onOff],
                [0x66, 'state_l8', tuya.valueConverter.onOff],
				[0x67, 'state_l9', tuya.valueConverter.onOff],
				[0x68, 'state_l10', tuya.valueConverter.onOff],
				[0x69, 'state_l11', tuya.valueConverter.onOff],
				[0x6A, 'state_l12', tuya.valueConverter.onOff],
            ],
    },
  whiteLabel: [
          tuya.whitelabel('ZYXH', 'TS0601_switch_12', '12 Gang switch', ['_TZE204_dqolcpcp']),
       ],
};

module.exports = definition;

**Reply by Koenkk** [reply]
Please create a pr by clicking the pencil icon here: https://github.com/Koenkk/zigbee-herdsman-converters/blob/master/src/devices/tuya.ts
